jupyter_server/services/contents/handlers.py

```diff
@@ -136,26 +137,41 @@ async def get(self, path=""):
             raise web.HTTPError(400, "Content %r is invalid" % content_str)
         content = int(content_str or "")
 
-        md5_str = self.get_query_argument("md5", default="0")
-        if md5_str not in {"0", "1"}:
-            raise web.HTTPError(400, "Content %r is invalid" % md5_str)
-        md5 = int(md5_str or "")
+        hash_str = self.get_query_argument("hash", default="0")
+        if hash_str not in {"0", "1"}:
+            raise web.HTTPError(400, f"Content {hash_str!r} is invalid")
+        require_hash = int(hash_str)
 
         if not cm.allow_hidden and await ensure_async(cm.is_hidden(path)):
             await self._finish_error(
                 HTTPStatus.NOT_FOUND, f"file or directory {path!r} does not exist"
             )
+
         try:
-            model = await ensure_async(
-                self.contents_manager.get(
-                    path=path,
-                    type=type,
-                    format=format,
-                    content=content,
-                    md5=md5,
+            expect_hash = require_hash
+            try:
+                model = await ensure_async(
+                    self.contents_manager.get(
+                        path=path,
+                        type=type,
+                        format=format,
+                        content=content,
+                        require_hash=require_hash,
+                    )
                 )
-            )
-            validate_model(model, expect_content=content, expect_md5=md5)
+            except TypeError:
+                # Fallback for ContentsManager not handling the require_hash argument
```

Comment:
This is a good fallback method. I had thought about using `inspect.signature` and I just want to get your opinion.

Reply:
I thought about doing that once. But as the handler are instantiated for each request, I think the try/except is faster than signature inspection.